### PR TITLE
✨ Enable Scorecard badge

### PR DESCRIPTION
**Pull request by azeemshaikh38**
#### Summary

Enable the beta version of scorecard-action:v2 which adds a Scorecard badge to the repo.

#### Release Note

NONE
#### Documentation
NONE

### CODE REVIEW

File: .github/workflows/scorecards-analysis.yml
Comment (woodruffw) [inline .github/workflows/scorecards-analysis.yml:23]: Why does action need this permission?

Comment (azeemshaikh38) [reply .github/workflows/scorecards-analysis.yml:23]: We use GitHub's OIDC token to sign the results which ensures integrity of the result displayed in the badge.

Comment (woodruffw) [reply .github/workflows/scorecards-analysis.yml:23]: Would you mind adding a comment to that effect? I realize the other permissions aren't documented, but it'd be nice to (eventually) have them all documented so that future maintainers understand the security scope of the workflow.

Comment (azeemshaikh38) [reply .github/workflows/scorecards-analysis.yml:24]: Done.

**Comment by woodruffw** [comment]
LGTM structurally, although I have a (weak) preference for waiting for v2 to be stabilized and officially released before we merge it here. 

cc @di however; if you're content as-is then I have no objections 🙂

File: .github/workflows/scorecards-analysis.yml
Comment (woodruffw) [reply .github/workflows/scorecards-analysis.yml:24]: Thanks! I'll admit that I'm still a little murky on what this means -- does it mean that the Scorecard server is verifying the OIDC-minted JWT to confirm that the results are actually being created from a particular run of this workflow?

Comment (azeemshaikh38) [reply .github/workflows/scorecards-analysis.yml:24]: Yep, that's correct.

**Comment by azeemshaikh38** [comment]
> LGTM structurally, although I have a (weak) preference for waiting for v2 to be stabilized and officially released before we merge it here.
> 
> cc @di however; if you're content as-is then I have no objections 🙂

If it helps, this version is already installed and running on [cosign](https://github.com/sigstore/cosign/blob/72c42246eecb4cfcb35e2dda9a2debbe5138ac8e/.github/workflows/scorecard_action.yml#L31), [rekor](https://github.com/sigstore/rekor/blob/0140a44cc063ab3f08884f685dd36efadd719570/.github/workflows/scorecard_action.yml#L31) and [fulcio](https://github.com/sigstore/fulcio/blob/0b23eb66b00690fbd8469c82e8bb397077b7421e/.github/workflows/scorecard_action.yml#L31) :)

**Review by di — APPROVED** [review]

File: .github/workflows/scorecards-analysis.yml
Comment (woodruffw) [reply .github/workflows/scorecards-analysis.yml:24]: Thanks for clarifying!

**Comment by di** [comment]
Looks like https://api.securityscorecards.dev/projects/github.com/sigstore/sigstore-python is 404? Is that expected?

**Comment by woodruffw** [comment]
> Looks like https://api.securityscorecards.dev/projects/github.com/sigstore/sigstore-python is 404? Is that expected?

I think it was waiting for the action to finish; it returns JSON for me now.

**Pull request merged**
